File: app/src/main/java/com/jgm90/cloudmusic/utils/DbHelper.java
package com.jgm90.cloudmusic.utils;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

import com.jgm90.cloudmusic.tables.PlaylistsTable;
import com.jgm90.cloudmusic.tables.SongsTable;

public class DbHelper extends SQLiteOpenHelper {

    private static final String DB_NAME = "cloudmusic.sqlite";
    private static final int DB_VERSION = 1;
    private static DbHelper instance;

    public DbHelper(Context context) {
        super(context, DB_NAME, null, DB_VERSION);
    }

    public static synchronized DbHelper getInstance(Context context) {
        if (instance == null) {
            instance = new DbHelper(context.getApplicationContext());
        }
        return instance;
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        PlaylistsTable.create(db);
        SongsTable.create(db);
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
    }
}

File: app/src/main/java/com/jgm90/cloudmusic/utils/DbHelper.kt
package com.jgm90.cloudmusic.utils

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper
import com.jgm90.cloudmusic.tables.PlaylistsTable
import com.jgm90.cloudmusic.tables.SongsTable

class DbHelper private constructor(context: Context) : SQLiteOpenHelper(context, DB_NAME, null, DB_VERSION) {
    override fun onCreate(db: SQLiteDatabase) {
        PlaylistsTable.create(db)
        SongsTable.create(db)
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
    }

    companion object {
        private const val DB_NAME = "cloudmusic.sqlite"
        private const val DB_VERSION = 1
        @Volatile
        private var instance: DbHelper? = null

        @JvmStatic
        @Synchronized
        fun getInstance(context: Context): DbHelper {
            if (instance == null) {
                instance = DbHelper(context.applicationContext)
            }
            return requireNotNull(instance)
        }
    }
}
